src/Passes/Source/RuleTransformationPass/Configuration.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+    /// Configuration class for the RuleTransformation pass.
+    class RuleTransformationPassConfiguration
+    {
+      public:
+        // Setup and construction
+        //
+
+        /// Setup function that attached the configuration to the ConfigurationManager.
+        void setup(ConfigurationManager& config);
+
+        /// Creates a configuration where all functionality is disabled.
+        static RuleTransformationPassConfiguration disable();
```

Comment:
(Based on the comment) Consider renaming `disable()` to something like `createDisabled()`.

Reply:
Fixed.